From 9d6f8db054a24151b8cd205f0853f6e1287d8d6f Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Mon, 28 Apr 2025 20:30:49 +0300
Subject: [PATCH] [SW-223026] nic: add lane taps to cpucp_nic_info

Add lane taps data to cpucp_nic_info in form,
that is complied with LKD requirements

Change-Id: I2f327397c2efbae60148ac11efab650a8dbbae44
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/cpucp_if.h b/cpucp_if.h
index 9630937..8c71796 100644
--- a/cpucp_if.h
+++ b/cpucp_if.h
@@ -2971,13 +2971,7 @@
 	UNKNOWN_SERDES_TYPE = 0xFFFF	/* serdes_type is u16 */
 };
 
-struct hl_lane_tap {
-	unsigned int tap	: 22;
-	unsigned int reserved	: 10;
-};
-
-#define HL_OAM_LANE_TAPS	48
-
+#define CPUCP_NIC_DEFAULT_LANES_NUM	48
 struct cpucp_nic_info {
 	struct cpucp_mac_addr mac_addrs[CPUCP_MAX_NICS];
 	uint64_t link_mask[CPUCP_NIC_MASK_ARR_LEN];
@@ -2986,18 +2980,20 @@
 	uint64_t link_ext_mask[CPUCP_NIC_MASK_ARR_LEN];
 	uint8_t qsfp_eeprom[CPUCP_NIC_QSFP_EEPROM_MAX_LEN];
 	uint64_t auto_neg_mask[CPUCP_NIC_MASK_ARR_LEN];
+	uint16_t serdes_type; /* enum cpucp_serdes_type */
 	union {
 		struct {
-			uint16_t serdes_type; /* enum cpucp_serdes_type */
 			uint16_t tx_swap_map[CPUCP_MAX_NICS];
-			uint8_t align_swaps[6];
+			uint8_t align_tx_swap_map[6];
 		};
 		struct {
-			uint16_t serdes_type_2; /* enum cpucp_serdes_type */
-			uint8_t align_serdes_type[6];
-			struct hl_lane_tap lane_taps[HL_OAM_LANE_TAPS];
-			uint8_t align_taps[CPUCP_MAX_NICS * sizeof(uint16_t) -
-						HL_OAM_LANE_TAPS * sizeof(struct hl_lane_tap)];
+			uint8_t align_serdes_type[2];
+			uint8_t pre3[CPUCP_NIC_DEFAULT_LANES_NUM];
+			uint8_t pre2[CPUCP_NIC_DEFAULT_LANES_NUM];
+			uint8_t pre1[CPUCP_NIC_DEFAULT_LANES_NUM];
+			uint8_t c0[CPUCP_NIC_DEFAULT_LANES_NUM];
+			uint8_t post1[CPUCP_NIC_DEFAULT_LANES_NUM];
+			uint8_t align_lane_taps[20];
 		};
 	};
 };
